docs/specs/xref.yml

```diff
@@ -390,3 +390,123 @@ outputs:
+      "description": "description for b",
+      "summary": "Link to @a?displayProperty=summary",
+    }
+outputs:
+  docs/b.json: |
+    { "content":{"name":"name for b","description":"description for b"} }
+  build.log: |
+    ["error","circular-reference","Found circular reference","docs/a.md"]
+    ["error","circular-reference","Found circular reference"]
+---
+# SDP non-cyclic xref reference should be resolved
+# whose display properties are plain text
+inputs:
+  docfx.yml:
+  docs/a.json: |
+    {
+      "$schema": "https://raw.githubusercontent.com/dotnet/docfx/v3/schemas/TestData.json",
+      "uid": "a",
+      "name": "name for a",
+      "inlineDescription": "description for a",
+      "summary": "Link to @b?displayProperty=inlineDescription",
+    }
+  docs/b.json: |
+    {
+      "$schema": "https://raw.githubusercontent.com/dotnet/docfx/v3/schemas/TestData.json",
+      "uid": "b",
+      "name": "name for b",
+      "inlineDescription": "description for b",
+      "summary": "Link to @a",
+    }
+outputs:
+  docs/a.json: |
+    {
+      "content": 
+      {
+        "name": "name for a",
+        "summary": "<p>Link to <a href=\"b.json\">description for b</a></p>\n",
+        "inlineDescription": "description for a",
+        "uid": "a"
+      }
+    }
+  docs/b.json: |
+    {
+      "content": 
+      {
+        "name": "name for b",
+        "summary": "<p>Link to <a href=\"a.json\">name for a</a></p>\n",
+        "inlineDescription": "description for b",
+        "uid": "b"
+      }
+    }
+---
+# SDP cyclic xref reference
+# whose display properties referring to another uid
+inputs:
+  docfx.yml:
+  docs/a.json: |
+    {
+      "$schema": "https://raw.githubusercontent.com/dotnet/docfx/v3/schemas/TestData.json",
+      "uid": "a",
+      "name": "name for a",
+      "inlineDescription": "description for a",
+      "summary": "Link to @b?displayProperty=summary",
+    }
+  docs/b.json: |
+    {
+      "$schema": "https://raw.githubusercontent.com/dotnet/docfx/v3/schemas/TestData.json",
+      "uid": "b",
+      "name": "name for b",
+      "inlineDescription": "description for b",
+      "summary": "Link to @a?displayProperty=summary",
+    }  
+  docs/c.json: |
+    {
+      "$schema": "https://raw.githubusercontent.com/dotnet/docfx/v3/schemas/TestData.json",
+      "uid": "c",
+      "name": "name for c",
+    }
+  docs/d.json: |
+    {
+      "$schema": "https://raw.githubusercontent.com/dotnet/docfx/v3/schemas/TestData.json",
+      "xref": "c",
+    }
+outputs:
+  docs/c.json: | 
+    {
+      "content": 
+      {
+        "name": "name for c",
+        "uid": "c"
+      }
+    }
+  docs/d.json: |
+    {
+      "content": 
+      {
+        "xref": "c.json"
+      }
+    }
+  build.log: |
+    ["error","circular-reference","Found circular reference","docs/b.json"]
+    ["error","circular-reference","Found circular reference","docs/a.json"]
+    ["error","circular-reference","Found circular reference"]
```

Comment:
The last one is caused by `OutputXrefMap` since it does not have a reference file, it is treated as a different error. This could be solved once we have the dependency chain. The only way I can think of to get it now is to add a call stack to the building file. Can we do it in another PR since this one is getting big and long-living